x-pack/plugins/ml/public/application/data_frame_analytics/pages/job_map/components/cytoscape.tsx

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, {
+  CSSProperties,
+  useState,
+  useRef,
+  useEffect,
+  ReactNode,
+  createContext,
+  useCallback,
+} from 'react';
+import cytoscape from 'cytoscape';
+import { cytoscapeOptions } from './cytoscape_options';
+
+export const CytoscapeContext = createContext<cytoscape.Core | undefined>(undefined);
+
+interface CytoscapeProps {
+  children?: ReactNode;
+  elements: cytoscape.ElementDefinition[];
+  height: number;
+  style?: CSSProperties;
+  width: number;
+}
+
+function useCytoscape(options: cytoscape.CytoscapeOptions) {
+  const [cy, setCy] = useState<cytoscape.Core | undefined>(undefined);
+  const ref = useRef(null);
+
+  useEffect(() => {
+    if (!cy) {
+      setCy(cytoscape({ ...options, container: ref.current }));
+    }
+  }, [options, cy]);
+
+  // Destroy the cytoscape instance on unmount
+  useEffect(() => {
+    return () => {
+      if (cy) {
+        cy.destroy();
+      }
+    };
+  }, [cy]);
+
+  return [ref, cy] as [React.MutableRefObject<any>, cytoscape.Core | undefined];
```

Comment:
Please avoid using `any`, especially when you share the `ref` across functions. Should be `<HTMLElement | null>`